src/jbi/configuration.py

```diff
@@ -0,0 +1,84 @@
+import importlib
+from inspect import signature
+from types import ModuleType
+from typing import Any, Dict, Optional
+
+import yaml
+from yaml import Loader
+
+from src.app import environment
+
+settings = environment.get_settings()
+
+
+class ConfigError(Exception):
+    pass
+
+
+class ProcessError(Exception):
+    pass
+
+
+def get_yaml_configurations(
+    jbi_config_file: str = "config.yaml", process_only_enabled: bool = True
+) -> Dict[str, Dict]:
+
+    with open(jbi_config_file, encoding="utf-8") as file:
+        try:
+            file_data = file.read()
+            data = yaml.load(file_data, Loader)
+            validated_action_dict = process_actions(
+                action_configuration=data.get("actions"),
+                process_only_enabled=process_only_enabled,
+            )
+            return validated_action_dict
+        except (ValueError, ConfigError, yaml.YAMLError) as exception:
+            raise ProcessError("PROCESS ERROR: Errors exist.") from exception
+
+
+def process_actions(
+    action_configuration, process_only_enabled: bool = True
+) -> Dict[str, Dict]:
+    requested_actions = {}
+    for yaml_action_key, inner_action_dict in action_configuration.items():
+        # Check enabled state
+        enabled = inner_action_dict.get("enabled")
```

Comment:
Maybe add a comment or write it like this to hightlight the fact that is disabled by default
```suggestion
        enabled = inner_action_dict.setdefault("enabled", False)
```

Reply:
Moving enablement logic outside of configuration process--based on conversation from earlier.